Mailchimp subscribe to new post notification feature added

Sabotnik opened the pull request:
Should be working, did a manual test and it allowed me to sign up. The only design feature I am still having an issue with is getting the "Subscribe" button inline with the text. Not sure what CSS selector to use since MailChimp has some of it's own stuff in there. I was able to change the styling on the button to use Bootstrap btn properties though.

If you take a look at the MailChimp documentation and let me know what you think, I can always tweak the styling. http://kb.mailchimp.com/lists/signup-forms/css-hooks-for-customizing-forms?&_ga=1.135060659.303637878.1439232708

Sabotnik committed:
added MailChimp signup form minus the styles

Sabotnik committed:
styled the MailChimp signup form

Sabotnik commented:
Ah yes - @kenmazaika I also will need to change some info on MailChimp - currently using my house address since I don't have an address for Firehose Project (didn't show up on Google) and need a physical address in the emails that go out. MailChimp is hooked to my email address as well, could also use some sweet icons to use and color scheme stuff if you want it to look nicer. I'd like to leave it hooked up to my account for the time being for learning sake and then turn it over to the Chief Firehoser or whomever once we get the features the way we want them.

kenmazaika commented:
@Sabotnik awesome.  Can you give @colinrubbert the credentials to log into the firehose community mailchimp account?

kenmazaika commented:
ah cool.  makes sense.  let's spin up a new account, maybe with a firehosecommunity@gmail.com or something.

Firehose Office is located at: 

```
25 Broad Street, Boston MA, 02114
```

Sabotnik commented:
@kenmazaika I will make the switch for the address now, is firehosecommunity@gmail.com a real email yet? With MailChimp, you can have an email for the account access "admin" and a different one for the reply to and from for each email campaign. Right now we have a "RSS feed driven campaign" setup. I will share the credentials for sure.

Sabotnik commented:
@kenmazaika what phone number do you want for "office phone"?

kenmazaika commented:
Solid.  Right now firehosecommunity gmail account was just an idea I had.  We haven't set it up.

Our office phone number (which is setup to go to a google voice mailbox that forwards the message to my email address) is: `(857) 362-8078`

colinrubbert commented:
We should have community specific email address for sure and definitely not use or personal accounts. Can you add contributors to mail chimp? That could be an easy solution to get around handing out email/passwords.

colinrubbert commented:
@Sabotnik I'll do my best to take a look at this tonight on my machine.

Sabotnik commented:
I just signed up for the free MailChimp account and put as much Firehose Community info as I could. The only personal info on it right now is my name, email and phone number just because I needed to put something in the forms. I don't think you can add contributors with the free account but I will look into it.

Sabotnik commented:
@kenmazaika @colinrubbert once you review the code and when it gets merged (let me know of any changes of course), you should sign up for the updates to make sure everything works. I can only go so far into the process because my email address is already on the list for the email campaign (we have a subscriber list of "1" right now). It worked initially to set it up when I did the "send a test email" option on the MailChimp side of things and I have tested out the signup form on the site from my local server, which does currently work to take me to the signup form.

colinrubbert commented:
That would be a great approach for the community page otherwise I'm sure there are specific lines in the code that connect to the account, @kenmazaika could just give us the pertinent information that we need to switch in the code to make the accounts operated by the community email account.

colinrubbert commented on _includes/mailchimp.html:
```diff
@@ -0,0 +1,18 @@
+<!-- Begin MailChimp Signup Form -->
+<div id="mc_embed_signup">
+	<form action="//thefirehoseproject.us11.list-manage.com/subscribe/post?u=48d53d72a28bb1c0fca326b5a&amp;id=953d41e871" method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" class="validate" target="_blank" novalidate>
```
@Sabotnik @kenmazaika I'm guessing this is where the accounts settings for the official firehose community email account will need to be placed otherwise the code should be the same.

kenmazaika commented:
I could setup an email like: `community@thefirehoseproject.com` to forward to a list of addresses.  I could be included on there as well as anyone else who would want to be.  

Let's spin up a new mailchimp account for this.  The firehose mailchimp we use today I think should remain separate, since this is a community driven effort.

Thoughts?

colinrubbert commented:
I agree on keeping the accounts separate and a `community@thefirehoseproject.com` would be better than a gmail account. That way anyone who subscribes doesn't think they are subscribing to something that isn't legitimate. I think that line of code that I commented on will be the piece that we will need to change so we can keep all the effort @Sabotnik put into this.

kenmazaika commented:
who should `community@thefirehoseproject.com` forward to?  I have to set it up on namecheap's DNS and I can do so in a moment.

Sabotnik commented:
That works for me for the email address. 2 things:
1) the code commented on above is simply tied to email "campaign" managed on MailChimp, i just happened to call it that
2) The account I have on MailChimp isn't mine anyhow - I set it up for the community, just need to put a different name, number and email. I used to have a MailChimp account long ago but turned that one over previously as well. No need to create a separate account.

colinrubbert commented:
@kenmazaika you can point it to either mine or @Sabotnik email address so that we can set up the account and do the heavy lifting and once that stuff is all setup you can switch it to your account so you change passwords and secure it.

colinrubbert commented:
@Sabotnik @kenmazaika I just looked at my mailchimp account and I'm 99% sure that I can confirm the line of code that I marked earlier will be the only thing that needs to be changed to get it working how it should.

Sabotnik commented:
You can put my email on there, that works. I suggest once we merge the code (or before) we should hop on a hangout and we can take a look on the MailChimp side and figure out if that does it for changing that piece of code or, if not, what to change. Would that work? I think I'm a bit confused as to what you are driving at. I have to leave for work in a bit but I'll be around tomorrow morning/afternoon if that works for you Colin?

colinrubbert commented:
@Sabotnik that line of code is basically what tells mailchimp where to send the sign up to. Currently that line of code is pointing at your personal account, when we set up the community account we will need to figure out what our unique subscriber code is and replace it with the one in place already. From there we can tackle other issues if there be any, we won't merge until it's fully functional. I can pull your branch and test it out without having to merge it into the master branch of the community site.

kenmazaika commented:
k.  I'll add the three of us.

colinrubbert commented:
Perfect

Sabotnik commented:
I switched the account credentials to the community@thefirehoseproject.com and the regenerated the code for the sign-up form - it is the same as what is there already in mailchimp.html.

I went through the "mailing list" that is setup for "New Blog Posts" on MailChimp and changed the email address that will be listed in the From: field. Looks like MailChimp uses Merge fields and tags to bring in info and data independently. It doesn't look like that piece of code changes when you switch around details.

In order to resolve the issue and get this merged, I would like to hop on Hangouts with one of you if you have questions - it is entirely possible that I'm wrong and, if so, I would like to learn from it. If not, let's get this pull request done and get some subscribers! I will be available tomorrow 11-12am CST and again 2-4p CST - just hit me up on Slack or Twitter

colinrubbert commented:
@Sabotnik are you logging into mail chimp with your personal account still or re you handing over ownership to the community email account? I'm not familiar with their process. I probably won't be able to jump in a call but I can possibly answer questions via slack.

Sabotnik commented:
@colinrubbert i switched it over to community email account already

colinrubbert commented:
Did you give it a password? Can you sign in with the community email address?

Sabotnik commented:
I made the login "firehoseproject" and gave it a password but can easily change it to the email address with password too - MailChimp is surprisingly easy to change that stuff, thought I would kind of be locked into whatever I originally chose. What else would you want me to resolve in order to get this closed out? I can share the login with you via Slack or email and you can take a look from the MailChimp side if you like...

colinrubbert commented:
Sharing the login would be great, you can send @kenmazaika the credentials, I shouldn't need them for anything. Next what you should work on is getting it looking good on the site. There's still some work that needs to be done to make it looks like it fits in with the rest of the content. I played with it a bit yesterday and tweaked the CSS. It was coming along. I'll select the point in the code where you could better optimize it to make it fit with the rest of the page, this can be done by overriding mailchimp's css with bootstraps.

colinrubbert commented on _includes/mailchimp.html:
```diff
@@ -0,0 +1,18 @@
+	<form action="//thefirehoseproject.us11.list-manage.com/subscribe/post?u=48d53d72a28bb1c0fca326b5a&amp;id=953d41e871" method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" class="validate" target="_blank" novalidate>
+    	<div id="mc_embed_signup_scroll">
+			<label for="mce-EMAIL">Want to be notified when new content is added?</label>
+			<input type="email" value="" name="EMAIL" class="email" id="mce-EMAIL" placeholder="email address" required>
```
Line 5 can be completely removed. Line 6 can be modified to have the "Want to be notified when new content is added?" in the placeholder. 

```
<input type="email" value="" name="EMAIL" class="email" id="mce-EMAIL" placeholder="Want to be notified when new content is added?" required>
```

From there you will have to do some work figuring out how to make thinks look like they belong. Look into the CSS bootstrap class "form-control", that will change the input box to make it look more like it belongs.

colinrubbert commented:
@Sabotnik you chose to tackle probably the most difficult task on the list of issues. There will be some good amount of work to go into it and the styling is going to be the biggest hassle. Keep working on it and if you need help you could ask other students as well, I know we've got some good designers in the group. Just keep asking questions the way you are and remember we don't need to rush it out the door, lets make this look amazing! Big props for tackling this!
